src/main/java/com/glencoesoftware/bioformats2raw/Converter.java

```diff
@@ -157,139 +258,297 @@ public class Converter implements Callable<Void> {
       "--output-options=s3fs_path_style_access=true|... " +
       "might be useful for connecting to minio."
   )
-  private Map<String, String> outputOptions;
+  public void setOutputOptions(Map<String, String> options) {
+    outputOptions = options;
+  }
 
+  /**
+   * Define the number of resolutions in the generated pyramid.
+   * By default, the resolution count is calculated based upon
+   * the input image size. The resolution count includes the
+   * full-resolution image, so must be greater than 0.
+   *
+   * @param resolutions pyramid resolution count
+   */
   @Option(
     names = {"-r", "--resolutions"},
     description = "Number of pyramid resolutions to generate"
   )
-  private volatile Integer pyramidResolutions;
+  public void setResolutions(int resolutions) {
+    if (resolutions > 0) {
+      pyramidResolutions = resolutions;
+    }
+    else {
+      LOGGER.warn("Ignoring invalid resolution count: {}", resolutions);
+    }
+  }
 
+  /**
+   * Define a subset of input series indexes to convert.
+   * Series indexes begin at 0 and are the Bio-Formats series/OME Image
+   * index reported when resolution flattening has not been applied.
+   *
+   * @param seriesToConvert series index list
+   */
   @Option(
     names = {"-s", "--series"},
     arity = "0..1",
     split = ",",
     description = "Comma-separated list of series indexes to convert"
   )
-  private volatile List<Integer> seriesList = new ArrayList<Integer>();
+  public void setSeriesList(List<Integer> seriesToConvert) {
+    if (seriesToConvert != null) {
+      seriesList = seriesToConvert;
+    }
+  }
 
+  /**
+   * Set the maximum tile width (X chunk size) for input and output.
+   *
+   * @param width tile width
+   */
   @Option(
     names = {"-w", "--tile_width"},
     description = "Maximum tile width (default: ${DEFAULT-VALUE}). " +
       "This is both the chunk size (in X) when writing Zarr and the tile " +
       "size used for reading from the original data. Changing the tile " +
-      "size may have performance implications."
+      "size may have performance implications.",
+    defaultValue = "1024"
   )
-  private volatile int tileWidth = 1024;
+  public void setTileWidth(int width) {
+    if (width > 0) {
+      tileWidth = width;
+    }
```

Comment:
Could we add a warning in case the value is ignored (similar to resolution above)?